project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Cetmix OU 2023
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0).
+from odoo import api, fields, models
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        track_visibility="onchange",
+        copy=False,
+        string="PR State",
+        compute="_compute_pr_state",
+        readonly=False,
+    )
+
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ("open", "Open"),
+            ("draft", "Draft"),
+            ("merged", "Merged"),
+            ("closed", "Closed"),
+        ]
+
+    @api.model
+    def create(self, vals):
+        new_rec = super(ProjectTaskState, self).create(vals)
+
+        # Set pr state if URI is provided at creation
+        if "pr_uri" in vals:
+            new_rec._set_pr_state(vals.get("pr_uri"))
+        return new_rec
+
+    def write(self, vals):
+        if "pr_uri" in vals:
+            self._set_pr_state(vals.get("pr_uri"), vals.get("pr_state"))
+        return super(ProjectTaskState, self).write(vals)
+
+    @api.depends("pr_uri", "project_id.pr_state_default")
+    def _compute_pr_state(self):
+        for task in self:
+            pr_uri = task.pr_uri
+            if not pr_uri:
+                task.pr_state = False
+            else:
+                if task.project_id and task.project_id.pr_state_default:
+                    task.pr_state = task.project_id.pr_state_default
+                else:
+                    ICPSudo = self.env["ir.config_parameter"].sudo()
+                    pr_state = ICPSudo.get_param(
+                        "project_task_pull_request_state.pr_state_default"
+                    )
+                    task.pr_state = pr_state
```

Comment:
What if no config value is defined or it is empty?

Reply:
resolved